### DISCUSSION TITLE
Router query = requestProvider.js.map ?

### DISCUSSION THREAD

**Discussion by GaddMaster** [vercel/next.js#35199] [Help]
Hi all.

When ever I use a router setup like so (In pages)

- /property/[code].jsx
- /contact/[code].jsx

I some times get a crazy query value which if relied upon will break the app.
Where I am always expecting code to be **abcde** it can sometime be **requestProvider.js.map**.
Its kind of annoying because I then need a line of code like so at the start of my component.
if (code === "requestProvider.js.map") return null;

Why is this and how to avoid ?

Daniel

**Comment by wubbahed** [comment]
I'm seeing a similar issue, but it seems to be triggered by having the Chrome DevTools open.

**Comment by GaddMaster** [comment]
Interesting observaction,  thanks !

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ghost** [comment]
> I'm seeing a similar issue, but it seems to be triggered by having the Chrome DevTools open.

We're saying the same issue. Triggered when DevTools is open. It occurs on all dynamic urls. But only if the Coinbase Wallet extension is enabled in the browser. Do yall have the cb wallet extension?

**Comment by ghost** [comment]
Solved. This is an issue with coinbase wallet. When the extension is enabled and chrome dev tools is open, the wallet extension requests `requestProvider.js.map`. Raising this bug internally with that team.

**Comment by ghost** [comment]
To avoid it in my code that was experiencing the same issue, I extended our global middleware with hte following
```js
// CB Wallet extension currently queries for `*/requestProvider.js.map` whenever the page loads and dev tools are open. it replaces the last
// path param with requestProvider.map.js the chrome dev tools is open. This causes errors in nextjs slug paths
// because it matches the url format but fails in the BE. 404 it here so we don't spam the BE.
if (req.url.match(/requestProvider\.js\.map$/)?.length) {
  const url = new URL(req.url);
  url.pathname = '/404';
  return NextResponse.rewrite(url.toString());
}
```
